### PR TITLE
CsvConverter: Support CSV basic options.

### PR DESCRIPTION
**Pull request by nirmo-lightricks**
Support different delimiters, escaping and quoting string.

### CODE REVIEW

File: requirements_dev.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ pandas==0.20.3
 PyYAML==3.11
 pytest==2.9.2
 pytest-runner==2.11.1
+six==1.10.0
```
Comment (c-nichols) [inline requirements_dev.txt]: This project uses the `future` library for a python3-first approach to compatibility -- use that instead for python2 support please

File: spectrify/convert.py [hunk 2]
```diff
@@ -141,8 +144,7 @@ def columnar_data_chunks(self, data_path, sa_table, chunk_size):
         # An array of functions corresponding to the CSV columns that take a string and return the
         # corresponding Python datatype
         type_converters = self.table_to_conversion_funcs(sa_table)
-
-        with S3GZipCSVReader(self.s3_config, data_path, delimiter='|', escapechar='\\', quoting=csv.QUOTE_NONE) as reader:
+        with S3GZipCSVReader(self.s3_config, data_path, delimiter=self.delimiter, escapechar=self.escapechar, quoting=self.quoting) as reader:
```
Comment (c-nichols) [inline spectrify/convert.py]: What do you think of putting the actual `S3GzipCSVReader()` instantiation into a helper function?  Would make it easier to override, and might be a bit easier to read

File: requirements_dev.txt (hunk 1 above)
Comment (nirmo-lightricks) [reply requirements_dev.txt]: done 👍

**Comment by nirmo-lightricks** [comment]
Hi @c-nichols,
I made all the changes, but for some reason I get the following error (see travis log for Python 3.4):

```python
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-viw7b8i2/pyarrow/setup.py", line 29, in <module>
        from Cython.Distutils import build_ext as _build_ext
    ImportError: No module named 'Cython'
    
```

do you know how to solve it?
Thanks :)

**Comment by c-nichols** [comment]
Thanks Nir!  I will take a look... I'm not sure off the top of my head.  Possible that the install process for Arrow has changed.

**Comment by c-nichols** [comment]
@nirmo-lightricks if you rebase master, the tests should all pass (dropped support for Python 3.4)

**Comment by nirmo-lightricks** [comment]
@c-nichols done :)

**Pull request merged**